ImagineEngine.xcodeproj/project.pbxproj

```diff
@@ -877,6 +880,51 @@
 		};
 /* End PBXResourcesBuildPhase section */
 
+/* Begin PBXShellScriptBuildPhase section */
+		C8F522E11FA8B937006B406B /* Run SwiftLint */ = {
+			isa = PBXShellScriptBuildPhase;
+			buildActionMask = 2147483647;
+			files = (
+			);
+			inputPaths = (
+			);
+			name = "Run SwiftLint";
+			outputPaths = (
+			);
+			runOnlyForDeploymentPostprocessing = 0;
+			shellPath = /bin/sh;
```

Comment:
Let's see... which target is this for? (hard to tell from the Xcode project diff 😅), I'm just thinking that we don't want to require SwiftLint for people who just want to install Imagine Engine and not work on it. Maybe run it only for the test targets, or run it only on CI by adding it to `buddybuild_postbuild.sh`? 🤔

Reply:
A build phase named "Run SwiftLint" was added to 3 targets: iOS, macOS, tvOS with the same contents. Totally agree that it is not clear what the contents are (read below for more info on this).

> I'm just thinking that we don't want to require SwiftLint for people who just want to install Imagine Engine and not work on it

That's a good thing to keep in mind, indeed! Strictly speaking though we do not require it: if the tool is absent on a user's machine, we will just show a warning.
![image](https://user-images.githubusercontent.com/755084/32384366-44ec1bbc-c0ed-11e7-80d4-601796b21127.png)

But we can go further and enable linting only for debug configurations. Something like that is being used in [Carthage codebase](https://github.com/Carthage/Carthage/blob/master/Carthage.xcodeproj/project.pbxproj#L738):

```
if [ "${CONFIGURATION}" = "Debug" ]; then
  # Linter code goes here
fi
```

Also to make it more readable, I'd suggest extracting that script to separate file and call it from the targets where we want to include linter. We can create something like `Scripts/run_swiftlint.sh` and invoke it from the build phases. It will fix the problem where we need to read that project file to get idea about what's going on in the script.

>Maybe run it only for the test targets

It is possible to do so, though usually I've seen adding the build phase to the main target. If you feel more comfortable adding this to test targets only, I'll do so 😉

>run it only on CI by adding it to buddybuild_postbuild.sh

Technically it is possible as well. The only thing that comes in mind for such an option is that we'll loose that integration with Xcode during the development cycle itself, and we will be forced to run the linter manually every time before commit (or somewhere near), it can be just a bit less convenient to do. And we will need to parse results somehow to present more detailed info in the PR...

...and it reminds me of Danger which is also a really neat tool to have installed for open source projects 😃 As the amount of contributions grow, we can think about setting it up somewhere in future 😉

As of now I'd suggest making limiting linter for debug configurations and extracting the script to separate file. Let me know what do you think about all this 😅

Reply:
I think adding SwiftLint to the test target strikes a good balance between being able to easily run it locally and also not being intrusive for users that don't have SwiftLint installed. Even if it's just a warning it's a bit of an eyesore 😅

Since we all run tests anyway as part of our dev flow, and all PRs so far have been accompanied by tests (which is great!), I think it should give us enough local feedback before submitting a PR 🙂

Reply:
Done! `Run SwiftLint` build phases have been moved to the corresponding test targets 🚀